Static checks are failing

```
Run 'ruff' for extremely fast Python linting................................Failed
- hook id: ruff
- exit code: 1

All checks passed!
providers/cncf/kubernetes/tests/unit/cncf/kubernetes/operators/test_pod.py:1040:16: E712 Avoid equality comparisons to `False`; use `if not pod.spec.automount_service_account_token:` for false 
```

--- a/kubernetes-tests/tests/kubernetes_tests/test_kubernetes_pod_operator.py
+++ b/kubernetes-tests/tests/kubernetes_tests/test_kubernetes_pod_operator.py
@@ -134,7 +134,6 @@ class TestKubernetesPodOperatorSystem:
             },
             "spec": {
                 "affinity": {},
-                "automountServiceAccountToken": True,
                 "containers": [
                     {
                         "image": "ubuntu:16.04",

--- a/providers/cncf/kubernetes/src/airflow/providers/cncf/kubernetes/operators/pod.py
+++ b/providers/cncf/kubernetes/src/airflow/providers/cncf/kubernetes/operators/pod.py
@@ -305,7 +305,7 @@ class KubernetesPodOperator(BaseOperator):
         node_selector: dict | None = None,
         image_pull_secrets: list[k8s.V1LocalObjectReference] | None = None,
         service_account_name: str | None = None,
-        automount_service_account_token: bool = True,
+        automount_service_account_token: bool | None = None,
         hostnetwork: bool = False,
         host_aliases: list[k8s.V1HostAlias] | None = None,
         tolerations: list[k8s.V1Toleration] | None = None,

--- a/providers/cncf/kubernetes/tests/unit/cncf/kubernetes/operators/test_pod.py
+++ b/providers/cncf/kubernetes/tests/unit/cncf/kubernetes/operators/test_pod.py
@@ -973,6 +973,7 @@ class TestKubernetesPodOperator:
                 foo: bar
             spec:
               serviceAccountName: foo
+              automountServiceAccountToken: false
               affinity:
                 nodeAffinity:
                   requiredDuringSchedulingIgnoredDuringExecution:
@@ -1036,6 +1037,8 @@ class TestKubernetesPodOperator:
         assert pod.spec.containers[0].image_pull_policy == "Always"
         assert pod.spec.containers[0].command == ["something"]
         assert pod.spec.service_account_name == "foo"
+        assert pod.spec.automount_service_account_token == False
+
         affinity = {
             "node_affinity": {
                 "preferred_during_scheduling_ignored_during_execution": [

--- a/providers/cncf/kubernetes/tests/unit/cncf/kubernetes/operators/test_pod.py
+++ b/providers/cncf/kubernetes/tests/unit/cncf/kubernetes/operators/test_pod.py
@@ -1037,7 +1037,7 @@ class TestKubernetesPodOperator:
         assert pod.spec.containers[0].image_pull_policy == "Always"
         assert pod.spec.containers[0].command == ["something"]
         assert pod.spec.service_account_name == "foo"
-        assert pod.spec.automount_service_account_token == False
+        assert not pod.spec.automount_service_account_token
 
         affinity = {
             "node_affinity": {

